### PR TITLE
Bugfix/issue 54 taken null

### LINKED ISSUE #54: score isn't showing correctly if no tricks taken
if the user takes 0 tricks, the score isn't updated as taken is still null

### PR DESCRIPTION
**Pull request by creisle**
bugfixes: 
- store name for dealer not index
- always store taken (was not for jester and wizard)
- use zero when taken is not defined (only set to a number in game state for the trick winners)

### CODE REVIEW

File: src/components/ScoreBoard/index.tsx [hunk 1]
```diff
@@ -30,12 +30,12 @@ function getRoundScore(
 ) {
     let score = null;
 
-    const { bet, taken } = scores[roundNumber]?.[playerId] || {};
+    const { bet, taken = 0 } = scores[roundNumber]?.[playerId] || {};
```
Comment (areisle) [inline src/components/ScoreBoard/index.tsx:33]: not sure if it ever is, but I have taken typed as number or null so setting default wouldn't work

**Review by areisle — CHANGES_REQUESTED** [review]

File: src/components/ScoreBoard/index.tsx (hunk 1 above)
Comment (creisle) [reply src/components/ScoreBoard/index.tsx:33]: k, will also cast then

Comment (areisle) [reply src/components/ScoreBoard/index.tsx:33]: now bets other peoples bets are visible to players who haven't bet yet

Comment (areisle) [reply src/components/ScoreBoard/index.tsx:33]: also score is being calculated before trick has been played messing with the overall scores

**Review by areisle — CHANGES_REQUESTED** [review]

**Pull request closed without merging**